File: src (cpp)/animal/aves/wild_colibri.cpp
#include "wild_colibri.h"

WildColibri::WildColibri(int _x, int _y, int _weight) : def_ID(21),
                                                        def_ratio_meat(0),
                                                        def_ratio_plant(60) {
  ID = def_ID;
  position.SetX(_x);
  position.SetY(_y);
  ratio_meat = def_ratio_meat;
  ratio_plant = def_ratio_plant;
  is_land_animal = true;
  is_water_animal = false;
  is_air_animal = true;
  weight = _weight;
  AddEnemy(2); // lion
  AddEnemy(3); // tiger
  AddEnemy(9); // python
  AddEnemy(19); // colibri
}
void WildColibri::Interact() {
  std::cout << "The wild colibri is going to skewer your eyes" << std::endl;
}

File: src/virtualzoo/animal/aves/WildColibri.java
package virtualzoo.animal.aves;

import virtualzoo.animal.behavior.BehaviorWild;
import virtualzoo.animal.diet.Herbivora;

/**
 * Kelas WildColibri mendefinisikan atribut untuk wild colibri.
 *
 * @author Rionaldi Chandraseta - 13515077
 * @version 3.0
 * @since 3.0
 */
public class WildColibri extends Aves {

  /**
   * Constructor
   *
   * Menciptakan wild colibri dengan posisi (locX, locY) dan berat animalWeight.
   *
   * @param locX Nilai absis posisi.
   * @param locY Nilai ordinat posisi.
   * @param animalWeight Nilai berat wild colibri.
   */
  public WildColibri(int locX, int locY, int animalWeight) {
    super();
    animalBehavior = new BehaviorWild();
    animalDiet = new Herbivora(animalWeight, 60);
    ID = 21;
    position.setLocation(locX, locY);
    isLandAnimal = true;
    isWaterAnimal = false;
    isAirAnimal = true;
    animalBehavior.addEnemy(2);
    animalBehavior.addEnemy(3);
    animalBehavior.addEnemy(9);
    animalBehavior.addEnemy(19);
  }

  /**
   * Mengatur interaksi dengan wild colibri.
   */
  public void interact() {
    System.out.println("The wild colibri is going to skewer your eyes");
  }
}
